lib/application.ts

```diff
@@ -708,7 +709,7 @@ export class SNApplication {
     return this.protocolService!.getDefaultOperatorEncryptionDisplayName();
   }
 
-  public async getUserVersion() {
+  public getUserVersion() {
```

Comment:
Wait why remove `async` here? `getUserVersion` is async.

Reply:
An async function is just a function that returns a promise. When we say:
```ts
public async getUserVersion() {
  return this.protocolService!.getUserVersion();
}
```

We're saying
```ts
public getUserVersion() {
  return new Promise(resolve => {
    this.protocolService!.getUserVersion().then(resolve);
  });
}
```

This is redundant and potentially costs us an extra event loop cycle (I'm not 100% but I believe the spec dictates Promises resolve on the next tick).
The async modifier should thus only be used for functions that need to use the await keyword. You can absolutely call an async function inside a non-async function and treat its result like the promise that it it, you just can't use await on that promise.

Reply:
Hmm..I usually read code as if it were documentation sometimes, because really we don't have docs for SNJS. Very often I check a function's signature to see how I should call it. Here I would probably look at this and say, oh cool, this function is sync, and I would call it as such in a unit test (where we don't have TS).

Reply:
How about this then?
```
public getUserVersion(): Promise<ProtocolVersion | undefined> {
  return this.protocolService!.getUserVersion();
}
```

Reply:
I suppose that works..